apps/web/src/server/applications/case/documentation/applications-documentation.controller.js

```diff
@@ -487,3 +494,67 @@ const getPaginationButtonData = (currentPageNumber, pageCount) => {
 		}))
 	};
 };
+
+/**
+ * Search documents.
+ *
+  @type {import('@pins/express').RenderHandler<ApplicationsSearchResultsProps, {}, ApplicationsSearchResultsBody, {q: string}, {pageNumber: string}>} */
+export async function searchDocuments(req, response) {
+	const { errors, body, params } = req;
+	const { query: bodyQuery } = body;
+
+	const query = bodyQuery ?? req.query.q;
+
+	const role = response.locals.domainType;
+	console.log('searchDocumnetcontroller');
+	if (errors || !query) {
+		return response.render(
+			'applications/case/case-documentation/search-results/document-search-results',
+			{
+				errors: errors || {},
+				searchApplicationsItems: [],
+				itemCount: 0,
+				query
+			}
+		);
+	}
+	console.log('passed error code');
+	const pageSize = 50;
+	const pageNumber = Number.parseInt(params?.pageNumber, 10) || 1;
+
+	const searchResponse = await applicationsDocumentationService.searchDocuments({
+		role,
+		query,
+		pageSize,
+		pageNumber
+	});
+	//console.log("Search Reponse:", searchResponse);
+	const searchApplicationsItems = searchResponse?.items || [];
```

Comment:
the API request could go wrong (a backend error). make sure you handle this somewhere